js/coinsbit.js

```diff
@@ -0,0 +1,742 @@
+            'order': undefined,
+            'type': undefined,
+            'takerOrMaker': takerOrMaker,
+            'side': side,
+            'price': price,
+            'amount': amount,
+            'cost': undefined,
+            'fee': undefined,
+        }, market);
+    }
+
+    sign (path, api = ['spot', 'public'], method = 'GET', params = {}, headers = undefined, body = undefined) {
+        const [ section, access ] = api;
+        let url = this.urls['api'][section][access] + 'v' + this.version + '/' + access + '/' + path;
+        // const defaultType = this.safeString (params, 'defaultType', 'spot'); defaultType same as section
+        params = this.omit (params, 'defaultType');
+        params = this.omit (params, this.extractParams (path));
+        if (access === 'public') {
+            if (method === 'GET') {
+                if (Object.keys (params).length) {
+                    url += '?' + this.urlencode (params);
+                }
+            }
+        } else {
+            this.checkRequiredCredentials ();
+            const timestamp = this.milliseconds ().toString ();
+            let auth = '';
+            headers = {
+                'ApiKey': this.apiKey,
+                'Request-Time': timestamp,
+                'Content-Type': 'application/json',
+            };
+            if (method === 'POST') {
+                auth = this.json (params);
+                body = auth;
+            } else {
+                params = this.keysort (params);
+                if (Object.keys (params).length) {
+                    auth += this.urlencode (params);
+                    url += '?' + auth;
+                }
+            }
+            auth = this.apiKey + timestamp + auth;
+            const signature = this.hmac (this.encode (auth), this.encode (this.secret), 'sha256');
+            headers['Signature'] = signature;
+        }
+        return { 'url': url, 'method': method, 'body': body, 'headers': headers };
+    }
+
+    handleErrors (code, reason, url, method, headers, body, response, requestHeaders, requestBody) {
+        //    {
+        //        success: true,
+        //        message: "",
+        //        code: "200",
+        //        result: [ .... ]
+        //    }
+        const success = this.safeValue (response, 'success');
+        if (success !== true) {
+            // depth endpoint doesn't return any code, so have to manually check.
+            const isBidAsks = url.includes ('depth/result');
+            if (isBidAsks && response && response['asks']) {
+                return;
+            }
+            // history/result endpoint doesn't return any code, so have to manually check.
+            const isHistoryResults = url.includes ('history/result');
+            if (isHistoryResults && response && Array.isArray (response)) {
+                return;
+            }
+            let errorMessage = '';
+            let responseCode = 0;
+            let feedback = this.id + ' ';
```

Comment:
The feedback message always defaults to `this.id + ' ' + body` in most of the exchanges.

Reply:
done